src/com/facebook/buck/go/GoTest.java

```diff
@@ -107,11 +107,12 @@ public boolean hasTestResultFiles(ExecutionContext executionContext) {
             ImmutableList.<String>builder()
                 .addAll(testMain.getExecutableCommand().getCommandPrefix(getResolver()))
                 .add("-test.v")
-                .add("-test.timeout", executionContext.getDefaultTestTimeoutMillis() + "ms")
+                .add(
+                    "-test.timeout",
+                    testRuleTimeoutMs.or(executionContext.getDefaultTestTimeoutMillis()) + "ms")
                 .build(),
             testMain.getExecutableCommand().getEnvironment(getResolver()),
             getPathToTestExitCode(),
-            testRuleTimeoutMs,
```

Comment:
What about adding some padding here and using both (in case the executable does hang)?